cli/src/main/java/com/devonfw/tools/ide/tool/LocalToolCommandlet.java

```diff
@@ -176,6 +176,73 @@ protected void postExtract(Path extractedDir) {
 
   }
 
+  @Override
+  public VersionIdentifier getInstalledVersion() {
+
+    return getInstalledVersion(this.context.getSoftwarePath().resolve(getName()));
+  }
+
+  /**
+   * @param toolPath the installation {@link Path} where to find the version file.
+   * @return the currently installed {@link VersionIdentifier version} of this tool or {@code null} if not installed.
+   */
+  protected VersionIdentifier getInstalledVersion(Path toolPath) {
+
+    if (!Files.isDirectory(toolPath)) {
+      this.context.debug("Tool {} not installed in {}", getName(), toolPath);
+      return null;
+    }
+    Path toolVersionFile = toolPath.resolve(IdeContext.FILE_SOFTWARE_VERSION);
+    if (!Files.exists(toolVersionFile)) {
+      Path legacyToolVersionFile = toolPath.resolve(IdeContext.FILE_LEGACY_SOFTWARE_VERSION);
+      if (Files.exists(legacyToolVersionFile)) {
+        toolVersionFile = legacyToolVersionFile;
+      } else {
+        this.context.warning("Tool {} is missing version file in {}", getName(), toolVersionFile);
+        return null;
+      }
+    }
+    try {
+      String version = Files.readString(toolVersionFile).trim();
+      return VersionIdentifier.of(version);
+    } catch (IOException e) {
+      throw new IllegalStateException("Failed to read file " + toolVersionFile, e);
+    }
+  }
+
+  @Override
+  public String getInstalledEdition() {
+
+    return getInstalledEdition(this.context.getSoftwarePath().resolve(getName()));
+  }
+
+  /**
+   * @param toolPath the installation {@link Path} where to find currently installed tool. The name of the parent
+   *     directory of the real path corresponding to the passed {@link Path path} must be the name of the edition.
+   * @return the installed edition of this tool or {@code null} if not installed.
+   */
+  public String getInstalledEdition(Path toolPath) {
+
+    if (!Files.isDirectory(toolPath)) {
+      this.context.debug("Tool {} not installed in {}", getName(), toolPath);
+      return null;
```

Comment:
I would still expect to get the configured edition if not installed.
The behavior of this code snipped can be considered as OK but then the `EditionGetCommandlet` also needs to be changed to handle `null` value and then print the configured edition.

Reply:
The EditionGetCommandlet already handles null values. When no installation is found, it prints the configured edition. https://github.com/devonfw/IDEasy/blob/79ff36f4f03126d465c164f72a52231e5f0c2931/cli/src/main/java/com/devonfw/tools/ide/commandlet/EditionGetCommandlet.java#L41